### PR TITLE
[ZK filter] Add support for setwatches2 ZK opcode

### PR DESCRIPTION
**Pull request by Winbobob**
Commit Message: [ZK filter] Add support for setwatches2 ZK opcode
Additional Description: Add support for setwatches2 ZK opcode: https://github.com/apache/zookeeper/blob/de7c5869d372e46af43979134d0e30b49d2319b1/zookeeper-jute/src/main/resources/zookeeper.jute#L83-L90
Risk Level: low
Testing: unit tests
Docs Changes: [Updates the ZK filter statistics](https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/network_filters/zookeeper_proxy_filter#statistics)

### CODE REVIEW

**Comment by phlax** [comment]
@JuniorHsu would you mind taking a look

@Winbobob this seems to be failing CI currently

**Comment by Winbobob** [comment]
> @JuniorHsu would you mind taking a look
> 
> @Winbobob this seems to be failing CI currently

Yeah, checking

**Comment by Winbobob** [comment]
cc @rgs1

**Review by JuniorHsu — APPROVED** [review]
lgtm

File: source/extensions/filters/network/zookeeper_proxy/decoder.h [hunk 1]
```diff
@@ -50,7 +50,8 @@ enum class OpCodes {
   SetAuth = 100,
   SetWatches = 101,
   GetEphemerals = 103,
-  GetAllChildrenNumber = 104
+  GetAllChildrenNumber = 104,
+  SetWatches2 = 105
```
Comment (Winbobob) [inline source/extensions/filters/network/zookeeper_proxy/decoder.h:54]: https://github.com/apache/zookeeper/blob/629518b5ea2b26d88a9ec53d5a422afe9b12e452/zookeeper-server/src/main/java/org/apache/zookeeper/ZooDefs.java#L92

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**